### PR TITLE
Run tests in CI on a big endian platform

### PR DESCRIPTION
**Pull request by psychon**
So far this is a bit WIP. I cannot properly test this due to #461, but I think this is basically done. I'll rewrite the git history after #461 is fixed.

I'm opening this PR already to give people a chance to tell me that I am Doing It Wrong (tm).

(Also, I am surprised how easy this was. Yay for rustup and cross.)

### CODE REVIEW

**Comment by psychon** [comment]
Updated. Commit https://github.com/psychon/x11rb/pull/462/commits/cd3d988fcec246e195a4863b98347edb9f80c106 causes a test failure only on the new build: https://travis-ci.org/github/psychon/x11rb/builds/695322947

**Comment by psychon** [comment]
Hrm. Besides test failing, it takes 12 minutes to run stuff for mips64. That's a lot more than the usual 8 to 9 minutes it takes for the other builds to finish. Do we want that on every CI run?

File: .travis.yml [hunk 1]
```diff
@@ -79,14 +104,21 @@ script:
...
+  - cargo test --verbose $FEATURES
+  - cargo doc --verbose $FEATURES
 
   # Run the examples as 'integration tests'.
   # Enable the 'all-extensions' feature or Cargo will complain.
   # Enable the 'allow-unsafe-code' feature so XCBConnection is used.
-  - run_examples --all-features
+  - skip_on_cross run_examples --all-features
 
   - cargo build --verbose --all-targets --features all-extensions
```
Comment (eduardosm) [inline .travis.yml:123]: This looks like it does `cargo build --verbose --all-targets --features all-extensions` twice.
Maybe MIPS CI could be reduced to just run parsing tests?

Comment (eduardosm) [reply .travis.yml:123]: > This looks like it does `cargo build --verbose --all-targets --features all-extensions` twice.

Well, that should not cause slowdown, the second time the command would just finish immediately.

Comment (psychon) [reply .travis.yml:123]: > Maybe MIPS CI could be reduced to just run parsing tests?

As in something like this? This now has `skip_on_cross` on everything but the call to `cargo test`.
```diff
diff --git a/.travis.yml b/.travis.yml
index d332c4c..b2ff68f 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -94,7 +94,7 @@ before_script:
 script:
   # Check the code generator. Output should be identical to the
   # committed version.
-  - make
+  - skip_on_cross make
   - if ! git diff --exit-code; then exit 1; fi
 
   - run_clippy
@@ -102,7 +102,7 @@ script:
 
   # Build once with just the default features, i.e. without all the extensions
   # to check that this works fine.
-  - cargo check --verbose --all-targets
+  - skip_on_cross cargo check --verbose --all-targets
 
   # Skip XCBConnection on cross (libxcb not available)
   - |
@@ -111,16 +111,16 @@ script:
       else
         export FEATURES="--features all-extensions"
       fi
-  - cargo build --verbose --all-targets $FEATURES
+  - skip_on_cross cargo build --verbose --all-targets $FEATURES
   - cargo test --verbose $FEATURES
-  - cargo doc --verbose $FEATURES
+  - skip_on_cross cargo doc --verbose $FEATURES
 
   # Run the examples as 'integration tests'.
   # Enable the 'all-extensions' feature or Cargo will complain.
   # Enable the 'allow-unsafe-code' feature so XCBConnection is used.
   - skip_on_cross run_examples --all-features
 
-  - cargo build --verbose --all-targets --features all-extensions
+  - skip_on_cross cargo build --verbose --all-targets --features all-extensions
 
   # Run the examples as 'integration tests'. This time using RustConnection.
-  - run_examples --features "all-extensions libc"
+  - skip_on_cross run_examples --features "all-extensions libc"
```

Comment (eduardosm) [reply .travis.yml:123]: Note that in Travis you can have an `include` entry use a specific `script`:

```yaml
    - env: CROSS_TARGET=mips64-unknown-linux-gnuabi64 CLIPPY_RUSTFMT=no
      rust: stable
      services: docker
      sudo: required 
      script:
        - cross test --target=...
```

Comment (psychon) [reply .travis.yml]: Nice, I didn't know that.

**Comment by psychon** [comment]
Rewrote this as suggested. Commit https://github.com/psychon/x11rb/pull/462/commits/68d4b64c7ac9ca7d30c84997da9b43c04f2e0350 again intentionally introduces an "endian error" and thus the corresponding travis build should fail: https://travis-ci.org/github/psychon/x11rb/builds/695360040
(Afterwards I git-force-pushed that commit away)

**Pull request merged**